float64 x_pos
float64 x_neg
float64 y_pos
float64 y_neg
float64 up
float64 down
